compiler/src/main/java/com/squareup/anvil/compiler/codegen/RealAnvilContext.kt

```diff
@@ -1,9 +1,21 @@
 package com.squareup.anvil.compiler.codegen
 
+import com.squareup.anvil.compiler.CommandLineOptions
 import com.squareup.anvil.compiler.api.AnvilContext
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
 
 internal data class RealAnvilContext(
   override val generateFactories: Boolean,
   override val generateFactoriesOnly: Boolean,
-  override val disableComponentMerging: Boolean
+  override val disableComponentMerging: Boolean,
+  override val module: ModuleDescriptor
 ) : AnvilContext
+
+internal fun CommandLineOptions.toAnvilContext(
+  module: ModuleDescriptor
+): RealAnvilContext = RealAnvilContext(
+  generateFactories = generateFactories,
+  generateFactoriesOnly = generateFactoriesOnly,
```

Comment:
General question: When do we only generate factories? Curious to understand the use-case

Reply:
That was actually a request from someone else. Maybe Slack? I don't remember. But we must support it now 🙃 

The idea is to generate all factories for Dagger to get rid of Kapt, but remove all other functionality from Anvil. It helps to convince teams to adopt Anvil.

Reply:
Gotcha, makes sense!